src/Mapbender/WmsBundle/Component/WmsInstanceConfigurationOptions.php

```diff
@@ -369,7 +393,75 @@ public static function fromArray($options)
             if (isset($options["exception_format"])) {
                 $ico->exceptionformat = $options["exception_format"];
             }
+            if (isset($options["layerOrder"])) {
+                $ico->setLayerOrder($options["layerOrder"]);
+            }
         }
         return $ico;
     }
+
+    /**
+     * Factory method, populates new instance from given entity
+     *
+     * @param WmsInstance $entity
+     * @return static
+     */
+    public static function fromEntity(WmsInstance $entity)
+    {
+        $options = new static();
+
+        $options->setUrl($entity->getSource()->getGetMap()->getHttpGet());
+        $dimensions = array();
```

Comment:
The function becomes really big. `getActiveDimensionConfigurations()` at this point?